lib/matplotlib/colorbar.py

```diff
@@ -1171,7 +1173,181 @@ def make_axes(parents, location=None, orientation=None, fraction=0.15,
+                layoutbox.match_heights([axpos, lbpos], [1., shrink])
+                layoutbox.align([axpos, lbpos], 'v_center')
+                # set the width of the pos box
+                lbpos.constrain_width(axpos.height * (1./aspect),
+                                      strength='strong')
+            elif location in ('bottom', 'top'):
+                lbpos = layoutbox.LayoutBox(
+                        parent=lb,
+                        name=lb.name + '.pos',
+                        tightheight=True,
+                        pos=True,
+                        subplot=False,
+                        artist=cax)
+
+                if location == 'bottom':
+                    layoutbox.vstack([axlb, lb], padding=0.01)
+                else:
+                    layoutbox.vstack([lb, axlb], padding=0.01)
+                # constrain the height and center...
+                layoutbox.match_widths([axpos, lbpos],
+                                       [1., shrink], strength='strong')
+                layoutbox.align([axpos, lbpos], 'h_center')
+                # set the height of the pos box
+                lbpos.constrain_height(axpos.width * (aspect),
+                                        strength='medium')
+        else:  # there is more than one parent, so lets use gridspec
+            # the colorbar will be a sibling of this gridspec, so the
+            # parent is the same parent as the gridspec.  Either the figure,
+            # or a subplotspec.
+            lb = layoutbox.LayoutBox(parent=gslb.parent,
+                                     name=gslb.parent.name + '.cbar',
+                                     artist=cax)
+            if location in ('left', 'right'):
+                lbpos = layoutbox.LayoutBox(
+                        parent=lb,
+                        name=lb.name + '.pos',
+                        tightwidth=False,
+                        pos=True,
+                        subplot=False,
+                        artist=cax)
+
+                if location == 'right':
+                    # arrange to right of the gridpec sibbling
+                    layoutbox.hstack([gslb, lb], padding=0.01,
+                                     strength='strong')
+                else:
+                    layoutbox.hstack([lb, gslb], padding=0.01)
+                # constrain the height and center...
+                # This isn't quite right.  We'd like the colorbar
+                # pos to line up w/ the axes poss, not the size of the
+                # gs.
+                maxrow = -100000
+                minrow = 1000000
+                maxax = None
+                minax = None
+
+                for ax in parents:
+                    subspec = ax.get_subplotspec()
+                    nrows, ncols = subspec.get_gridspec().get_geometry()
+                    for num in [subspec.num1, subspec.num2]:
+                        rowNum1, colNum1 = divmod(subspec.num1, ncols)
```

Comment:
I know tight_layout uses camelCase here and there but let's try to keep_using_underscores.